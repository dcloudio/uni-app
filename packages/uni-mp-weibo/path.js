const path = require('path')
module.exports = {
  src: path.join(__dirname, 'src'),
  lib: path.join(__dirname, 'lib')
}
