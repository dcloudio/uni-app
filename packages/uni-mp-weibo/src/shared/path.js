export function parsePath (path) {
  let hash = ''
  let query = ''

  const hashIndex = path.indexOf('#')
  if (hashIndex >= 0) {
    hash = path.slice(hashIndex)
    path = path.slice(0, hashIndex)
  }

  const queryIndex = path.indexOf('?')
  if (queryIndex >= 0) {
    query = path.slice(queryIndex + 1)
    path = path.slice(0, queryIndex)
  }

  return {
    path,
    query,
    hash
  }
}
