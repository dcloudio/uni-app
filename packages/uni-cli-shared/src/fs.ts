import fs from 'fs'
import path from 'path'

export function emptyDir(dir: string, skip: string[] = []) {
  for (const file of fs.readdirSync(dir)) {
    if (skip.includes(file)) {
      continue
    }
    // node >= 14.14.0
    fs.rmSync(path.resolve(dir, file), { recursive: true, force: true })
  }
}
