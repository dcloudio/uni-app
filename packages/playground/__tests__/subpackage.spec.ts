import fs from 'fs-extra'
import path from 'path'
import execa from 'execa'

const projectDir = path.resolve(__dirname, '../subpackage')

describe('subpackage playground', () => {
  jest.setTimeout(50 * 1000)
  const types = {
    'uni-app': ['dev:mp-weixin', 'build:mp-weixin'],
    'uni-app-x': ['dev:mp-weixin', 'build:mp-weixin'],
  }
  const distDir = path.resolve(projectDir, 'dist')
  if (fs.existsSync(distDir)) {
    fs.emptyDirSync(distDir)
  }
  Object.keys(types).forEach((type) => {
    const scripts = types[type]
    scripts.forEach((script) => {
      const mode = script.split(':')[0]
      const platform = script.split(':')[1]
      test(`${type} ${script}`, async () => {
        // 仅未指定输出目录时，分包产物位置会自动根据subpackage参数生成。此测试例中指定了UNI_OUTPUT_DIR，outDir也需要手动指定sub目录
        const outDir = path.resolve(distDir, mode, type, platform, 'sub')
        console.log(`${type} npm run ${script} start`)
        await execa('npm', ['run', script, '--', '--subpackage=sub'], {
          cwd: projectDir,
          env: {
            ...process.env,
            UNI_OUTPUT_DIR: outDir,
            UNI_APP_X: type === 'uni-app-x' ? 'true' : 'false',
          },
          stdio: 'inherit',
        })
        console.log(`${type} npm run ${script} end`)
        const files = [
          'common/assets.js',
          'pages/index/index.wxss',
          'uni_modules/test-module/pages/index/index.wxss',
          'uni_modules/test-module/components/test-component/test-component.wxss',
        ]
        files.forEach((file) => {
          const filePath = path.resolve(outDir, file)
          console.log(filePath)
          expect(fs.existsSync(filePath)).toBe(true)
          expect(fs.readFileSync(filePath, 'utf-8')).toMatchSnapshot(
            `${type} ${script} ${file}`
          )
        })
      })
    })
  })
})
