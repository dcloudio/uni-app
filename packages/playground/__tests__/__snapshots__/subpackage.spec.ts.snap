// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subpackage playground uni-app build:mp-weixin: uni-app build:mp-weixin common/assets.js 1`] = `
""use strict";
const _imports_0$1 = "/sub/static/logo.png";
const _imports_1$1 = "/sub/static/logo.png";
const _imports_2 = "/sub/assets/logo-dc.759742f5.png";
const _imports_0 = "/sub/uni_modules/test-component/static/logo-hx.png";
const _imports_1 = "/sub/uni_modules/test-module/static/logo-hx.png";
exports._imports_0 = _imports_0$1;
exports._imports_0$1 = _imports_0;
exports._imports_1 = _imports_1$1;
exports._imports_1$1 = _imports_1;
exports._imports_2 = _imports_2;
"
`;

exports[`subpackage playground uni-app build:mp-weixin: uni-app build:mp-weixin pages/index/index.wxss 1`] = `
"@import "../../app.wxss";
"
`;

exports[`subpackage playground uni-app build:mp-weixin: uni-app build:mp-weixin uni_modules/test-module/components/test-component/test-component.wxss 1`] = `""`;

exports[`subpackage playground uni-app build:mp-weixin: uni-app build:mp-weixin uni_modules/test-module/pages/index/index.wxss 1`] = `
"@import "../../../../app.wxss";
"
`;

exports[`subpackage playground uni-app dev:mp-weixin: uni-app dev:mp-weixin common/assets.js 1`] = `
""use strict";
const _imports_0$1 = "/sub/static/logo.png";
const _imports_1$1 = "/sub/static/logo.png";
const _imports_2 = "/sub/assets/logo-dc.759742f5.png";
const _imports_0 = "/sub/uni_modules/test-component/static/logo-hx.png";
const _imports_1 = "/sub/uni_modules/test-module/static/logo-hx.png";
exports._imports_0 = _imports_0$1;
exports._imports_0$1 = _imports_0;
exports._imports_1 = _imports_1$1;
exports._imports_1$1 = _imports_1;
exports._imports_2 = _imports_2;
"
`;

exports[`subpackage playground uni-app dev:mp-weixin: uni-app dev:mp-weixin pages/index/index.wxss 1`] = `
"@import "../../app.wxss";
"
`;

exports[`subpackage playground uni-app dev:mp-weixin: uni-app dev:mp-weixin uni_modules/test-module/components/test-component/test-component.wxss 1`] = `""`;

exports[`subpackage playground uni-app dev:mp-weixin: uni-app dev:mp-weixin uni_modules/test-module/pages/index/index.wxss 1`] = `
"@import "../../../../app.wxss";
"
`;

exports[`subpackage playground uni-app-x build:mp-weixin: uni-app-x build:mp-weixin common/assets.js 1`] = `
""use strict";
const _imports_0$1 = "/sub/static/logo.png";
const _imports_1$1 = "/sub/static/logo.png";
const _imports_2 = "/sub/assets/logo-dc.759742f5.png";
const _imports_0 = "/sub/uni_modules/test-component/static/logo-hx.png";
const _imports_1 = "/sub/uni_modules/test-module/static/logo-hx.png";
exports._imports_0 = _imports_0$1;
exports._imports_0$1 = _imports_0;
exports._imports_1 = _imports_1$1;
exports._imports_1$1 = _imports_1;
exports._imports_2 = _imports_2;
"
`;

exports[`subpackage playground uni-app-x build:mp-weixin: uni-app-x build:mp-weixin pages/index/index.wxss 1`] = `
"@import "../../app.wxss";
@import "../../uvue.wxss";
:host{display:flex;flex-direction:column}
"
`;

exports[`subpackage playground uni-app-x build:mp-weixin: uni-app-x build:mp-weixin uni_modules/test-module/components/test-component/test-component.wxss 1`] = `
":host{display:flex;flex-direction:column}
"
`;

exports[`subpackage playground uni-app-x build:mp-weixin: uni-app-x build:mp-weixin uni_modules/test-module/pages/index/index.wxss 1`] = `
"@import "../../../../app.wxss";
@import "../../../../uvue.wxss";
:host{display:flex;flex-direction:column}
"
`;

exports[`subpackage playground uni-app-x dev:mp-weixin: uni-app-x dev:mp-weixin common/assets.js 1`] = `
""use strict";
const _imports_0$1 = "/sub/static/logo.png";
const _imports_1$1 = "/sub/static/logo.png";
const _imports_2 = "/sub/assets/logo-dc.759742f5.png";
const _imports_0 = "/sub/uni_modules/test-component/static/logo-hx.png";
const _imports_1 = "/sub/uni_modules/test-module/static/logo-hx.png";
exports._imports_0 = _imports_0$1;
exports._imports_0$1 = _imports_0;
exports._imports_1 = _imports_1$1;
exports._imports_1$1 = _imports_1;
exports._imports_2 = _imports_2;
"
`;

exports[`subpackage playground uni-app-x dev:mp-weixin: uni-app-x dev:mp-weixin pages/index/index.wxss 1`] = `
"@import "../../app.wxss";
@import "../../uvue.wxss";
:host{display:flex;flex-direction:column}
"
`;

exports[`subpackage playground uni-app-x dev:mp-weixin: uni-app-x dev:mp-weixin uni_modules/test-module/components/test-component/test-component.wxss 1`] = `
":host{display:flex;flex-direction:column}
"
`;

exports[`subpackage playground uni-app-x dev:mp-weixin: uni-app-x dev:mp-weixin uni_modules/test-module/pages/index/index.wxss 1`] = `
"@import "../../../../app.wxss";
@import "../../../../uvue.wxss";
:host{display:flex;flex-direction:column}
"
`;
